core/src/voting_service.rs

```diff
@@ -0,0 +1,54 @@
+use crate::replay_stage::{VoteInfo, VoteOp};
+use solana_gossip::cluster_info::ClusterInfo;
+use solana_poh::poh_recorder::PohRecorder;
+use std::{
+    sync::{mpsc::Receiver, Arc, Mutex},
+    thread::{self, Builder, JoinHandle},
+};
+
+pub struct VotingService {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl VotingService {
+    pub fn new(
+        vote_receiver: Receiver<VoteInfo>,
+        cluster_info: Arc<ClusterInfo>,
+        poh_recorder: Arc<Mutex<PohRecorder>>,
+    ) -> Self {
+        let thread_hdl = Builder::new()
+            .name("sol-vote-service".to_string())
+            .spawn(move || {
+                for vote_info in vote_receiver.iter() {
+                    Self::push_vote(&cluster_info, &poh_recorder, vote_info);
+                }
+            })
+            .unwrap();
+        Self { thread_hdl }
+    }
+
+    pub fn push_vote(
```

Comment:
This `push_vote` name is a bit confusing, because then there is another `PushVote` and also a `RefreshVote`. Maybe rename this `handle_vote` or something.